Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/AddUserController.java

```diff
@@ -26,26 +28,41 @@
 	 * 
 	 * @param req
 	 *            Holds HTTP request information
+	 * @param model
+	 * 			  Holds generic information about the model
+	 * @param user
+	 * 			  Holds the User bean values retrieved from the add page
 	 * @return String to redirect user to user list page
 	 */
 	@RequestMapping(value = "auth/user/createuser")
-	public String createUser(HttpServletRequest req) {
+	public String createUser(HttpServletRequest req, ModelMap model, @ModelAttribute("user") User user) {
 
-		User user = new User(req.getParameter("username"),
-				req.getParameter("password"));
-		user.setName(req.getParameter("fullname"));
-		user.setEmail(req.getParameter("email"));
-		usersManager.addUser(user);
+		User newUser = new User(user.getUser(), user.getPw());
+		newUser.setName(user.getName());
+		newUser.setEmail(user.getEmail());
+		if(usersManager.findUser(newUser.getUser())== null)
+			usersManager.addUser(user);
+		else
+		{
+			model.addAttribute("errUserName", "Username already exists. Please select a new Username");
+			model.addAttribute("user", newUser);
+			return "auth/user/add";
+		}	
 		return "redirect:/auth/user/list";
 	}
 
 	/**
 	 * This method provides string to redirect user to add user page
 	 * 
+	 * @param model
+	 * 				Holds generic information about the model
+	 *            
+	 * 
 	 * @return String to redirect user to add user page
 	 */
 	@RequestMapping(value = "auth/user/add")
-	public String addUser() {
+	public String addUser(ModelMap model) {

```

Comment:
I know you didn't name this method, but let's call it something else. Maybe "preparePage" or something?